tools/snap/build_remote.py

```diff
@@ -103,6 +105,12 @@ def _build_snap(
         workspace = CERTBOT_DIR
     else:
         workspace = join(CERTBOT_DIR, target)
+        # Init and commit git repo in workspace. This is necessary starting in core24
+        # as "Projects must be at the top level of a git repository"
+        # https://snapcraft.io/docs/migrate-core24#remote-build
+        subprocess.run(['git', 'init'], capture_output=True, check=True, cwd=workspace)
+        subprocess.run(['git', 'add', '-A'], capture_output=True, check=True, cwd=workspace)
+        subprocess.run(['git', 'commit', '-m', 'init'], capture_output=True, check=True, cwd=workspace)
```

Comment:
i'd be surprised if it isn't, but do we know `git` is available in the build environment?

Reply:
hmmmmm this is a great question. I mean, how did you get a copy of the repo without git available? We do make the assumption that the azure has git available [here](https://github.com/certbot/certbot/blob/master/.azure-pipelines/templates/jobs/packaging-jobs.yml#L87), but this script can also be used manually.

Since `snapcraft remote-build` is in fact using git to upload the code, we could probably just grab the git packaged within `snapcraft`...but also, I'm not sure that's necessary. I could add a line to `tools/snap/README.md` mentioning that it's a requirement though if we really want?

Reply:
good points, also i just confirmed that it's present: https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md

imo that's sufficient!